src/components/shared/Button.jsx

```diff
@@ -0,0 +1,72 @@
+import styled, { css } from "styled-components";
+
+const BUTTON_SIZES = {
+  md: css`
+    --button-min-width: 240px;
+    --button-min-height: 64px;
+    --button-font-size: 1rem;
+    --button-padding: 12px 24px;
+    --button-border: 2px solid #000;
+    --button-radius: 4px;
+  `,
+  lg: css`
+    --button-min-width: 420px;
+    --button-min-height: 64px;
+    --button-font-size: 1rem;
+    --button-padding: 12px 24px;
+    --button-border: 4px solid #000;
+    --button-radius: 4px;
+  `,
+};
+
+const USING_CASES = {
+  next: css`
+    --button-color: #fff;
+    --button-background-color: #000;
+  `,
+  before: css`
+    --button-color: #000;
+    --button-background-color: #000;
+  `,
+};
+
+function Button({ disabled, handleClick, children, size, usingCase }) {
+  const buttonStyle = BUTTON_SIZES[size];
+  const buttonMode = USING_CASES[usingCase];
+
+  return (
+    <StyledButton
+      onClick={handleClick}
+      disabled={disabled}
+      $buttonStyle={buttonStyle}
+      $usedAt={buttonMode}
+    >
+      {children}
+    </StyledButton>
+  );
+}
+
+const StyledButton = styled.button`
+  ${props => props.$buttonStyle}
+  ${props => props.$usedAt}
+
+  min-width: var(--button-min-width);
+  min-height: var(--button-min-height);
+  margin: 0;
+  padding: var(--button-padding, 12px 16px);
+  border: var(--button-border);
+  border-radius: var(--button-radius, 8px);
+
+  font-size: var(--button-font-size, 1rem);
+  color: var(--button-color, #ffffff);
```

Comment:
P1: 해당 부분도 컨벤션과 동일하게 #fff 로 맞춰주시면 좋을 것 같습니다.

```suggestion
  color: var(--button-color, #fff);
```

Reply:
확인했습니다!
수정하겠습니다